devito/stencilkernel.py

```diff
@@ -29,11 +39,22 @@ def __init__(self, stencils, name="Kernel", compiler=None):
         stencils = stencils if isinstance(stencils, list) else [stencils]
         self.expressions = [Expression(s) for s in stencils]
 
+        # Lower all expressions to "indexed" API
+        for e in self.expressions:
+            e.indexify()
+
+        # Apply supplied substitutions
+        if subs is not None:
+            for expr in self.expressions:
+                expr.substitute(subs)
```

Comment:
As a paraphrase of this discussion, only to see if I get it right: am I correct if I say that once you pass an object to, say, an ``Iteration``, you basically lose it because it gets transformed in place?